spec/helper/MockHelper.ts

```diff
@@ -0,0 +1,335 @@
+      LIVE: 'live',
+      VOD: 'vod',
+      UNKNOWN: 'unknown',
+    };
+  }
+
+  export function getConvivaClientMock(): Conviva.Client {
+    const createSession = jest.fn(() => 0);
+    const cleanupSession = jest.fn();
+    const sendCustomEvent = jest.fn();
+    const updateContentMetadata = jest.fn();
+    const adStart = jest.fn();
+    const adEnd = jest.fn();
+    const reportError = jest.fn();
+    const releasePlayerStateManager = jest.fn();
+
+    const playerStateMock = getPlayerStateManagerMock();
+
+    global.Conviva.Client = jest.fn().mockImplementation(() => {
+      return {
+        sendCustomEvent,
+        getPlayerStateManager: jest.fn(() => playerStateMock),
+        releasePlayerStateManager,
+        createSession,
+        attachPlayer: jest.fn(),
+        detachPlayer: jest.fn(),
+        cleanupSession,
+        updateContentMetadata,
+        adStart,
+        adEnd,
+        reportError,
+      };
+    });
+
+    global.Conviva.Client.NO_SESSION_KEY = -2;
+
+    global.Conviva.Client.AdPosition = {
+      PREROLL: 'preroll',
+    };
+    global.Conviva.Client.AdStream = {
+      SEPARATE: 'separate',
+    };
+    global.Conviva.Client.AdPlayer = {
+      SEPARATE: 'separate',
+      CONTENT: 'content',
+    };
+    global.Conviva.Client.ErrorSeverity = {
+      FATAL: 'fatal',
+    };
+
+    return new global.Conviva.Client();
+  }
+
+  export function getPlayerStateManagerMock(): Conviva.PlayerStateManager {
+    const setPlayerState = jest.fn();
+    const setPlayerSeekStart = jest.fn();
+    const setPlayerSeekEnd = jest.fn();
+
+    const PlayerStateManagerClass = jest.fn().mockImplementation(() => ({
+      setPlayerType: jest.fn(),
+      setPlayerVersion: jest.fn(),
+      setPlayerState,
+      setPlayerSeekStart,
+      setPlayerSeekEnd,
+    }));
+
+    global.Conviva.PlayerStateManager = PlayerStateManagerClass;
+    global.Conviva.PlayerStateManager.PlayerState = {
+      STOPPED: 'stopped',
+      BUFFERING: 'buffering',
+      PAUSED: 'paused',
+      PLAYING: 'playing',
+    };
+
+    return new PlayerStateManagerClass();
+  }
+
+  export function getPlayerMock(): TestingPlayerAPI {
+    const eventHelper = new PlayerEventHelper();
+
+    const PlayerMockClass: jest.Mock<TestingPlayerAPI> = jest.fn().mockImplementation(() => {
+      return {
+        getSource: jest.fn(),
+        exports: { PlayerEvent },
+        getDuration: jest.fn(),
+        isLive: jest.fn(),
+        getConfig: jest.fn(() => {
+          return {};
+        }),
+        isPlaying: jest.fn(),
+        isPaused: jest.fn(),
+        getPlayerType: jest.fn(),
+        getStreamType: jest.fn(() => 'hls'),
+
+        // Event faker
+        eventEmitter: eventHelper,
+        on: eventHelper.on.bind(eventHelper),
+      };
+    });
+
+    return new PlayerMockClass();
+  }
+}
+
+export interface TestingPlayerAPI extends PlayerAPI {
+  eventEmitter: EventEmitter;
+}
+
+interface EventEmitter {
```

Comment:
This interface is not really needed, why do we have it?

Reply:
This make sure that we don't need a any cast when firing events.